src/ray/gcs/gcs_client/service_based_accessor.h

```diff
@@ -26,8 +26,31 @@ using SubscribeOperation = std::function<Status(const StatusCallback &done)>;
 
 using FetchDataOperation = std::function<void(const StatusCallback &done)>;
 
+using FetchDataOperation = std::function<void(const StatusCallback &done)>;
+
 class ServiceBasedGcsClient;
 
+/// \class IdempotentFilter
+/// `IdempotentFilter` is used to filter duplicate or older data.
+class IdempotentFilter {
+ public:
+  /// Check whether the data is duplicate or older. Non-thread safe.
```

Comment:
Does it mean the caller should always hold locks whenever it uses gcs client?

Reply:
> Does it mean the caller should always hold locks whenever it uses gcs client?

GCS client and the main thread use the same event loop, so they are actually called by a single thread.